File: web/src/api/models/Drops/DropDetailsDto.ts
import { DropItemDto } from './DropItemDto';

export interface DropDetailsDto {
  id: number;
  name: string;
  description: string;
  startDateTime: string;
  endDateTime: string;
  availableDropItems: DropItemDto[];
  reservedDropItems: DropItemDto[];
}

File: api/DropShot.Application/Drops/Models/DropDetailsDto.cs
using DropShot.Application.Common.AutoMapper;
using DropShot.Domain.Entities;

namespace DropShot.Application.Drops.Models;

public class DropDetailsDto : IMapFrom<Drop>
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime StartDateTime { get; set; }
    public DateTime EndDateTime { get; set; }

    public ICollection<DropItemDto> AvailableDropItems { get; set; }
    public ICollection<DropItemDto> ReservedDropItems { get; set; }
}